cmd/submit_test.go

```diff
@@ -184,6 +185,54 @@ func TestSubmitFiles(t *testing.T) {
 	assert.Equal(t, "This is the readme.", submittedFiles["README.md"])
 }
 
+func TestLegacySolutionMetadataMigration(t *testing.T) {
+	oldOut := Out
+	oldErr := Err
+	Out = ioutil.Discard
+	Err = ioutil.Discard
+	defer func() {
+		Out = oldOut
+		Err = oldErr
+	}()
+	// The fake endpoint will populate this when it receives the call from the command.
+	submittedFiles := map[string]string{}
+	ts := fakeSubmitServer(t, submittedFiles)
+	defer ts.Close()
+
+	tmpDir, err := ioutil.TempDir("", "legacy-metadata-file")
+	defer os.RemoveAll(tmpDir)
+	assert.NoError(t, err)
+
+	dir := filepath.Join(tmpDir, "bogus-track", "bogus-exercise")
+	os.MkdirAll(dir, os.FileMode(0755))
+	writeFakeLegacySolution(t, dir, "bogus-track", "bogus-exercise")
+
+	file := filepath.Join(dir, "file.txt")
+	err = ioutil.WriteFile(file, []byte("This is a file."), os.FileMode(0755))
+	assert.NoError(t, err)
+
+	v := viper.New()
+	v.Set("token", "abc123")
+	v.Set("workspace", tmpDir)
+	v.Set("apibaseurl", ts.URL)
+	cfg := config.Config{
+		Persister:       config.InMemoryPersister{},
+		Dir:             tmpDir,
+		UserViperConfig: v,
+	}
+	exercise := workspace.NewExerciseFromDir(dir)
+	expectedPathAfterMigration := exercise.MetadataFilepath()
+	_, err = os.Stat(expectedPathAfterMigration)
+	assert.Error(t, err)
+
+	err = runSubmit(cfg, pflag.NewFlagSet("fake", pflag.PanicOnError), []string{file})
+	assert.NoError(t, err)
+	assert.Equal(t, "This is a file.", submittedFiles["file.txt"])
+
+	_, err = os.Stat(expectedPathAfterMigration)
+	assert.NoError(t, err)
+}
```

Comment:
This looks good. Can we also check that there's no file at the `legacy` location?